modules/address/components/Avatar.tsx

```diff
@@ -24,25 +24,30 @@ export function Avatar({ size, address, defaultComponent, style }: AvatarProps):
   const { chainId, provider: library } = useWeb3();
   const [avatarUri, setAvatarUri] = useState<string | null>(null);
 
-  const fetchAvatarUri = async () => {
-    if (library && address && chainId !== SupportedChainId.GOERLIFORK) {
-      const avt = new AvatarResolver(library, { cache: 3600 /* cache for an hour */ });
-      try {
-        const ensName = await library.lookupAddress(address);
-        if (ensName) {
-          const uri = await avt.getAvatar(ensName, {});
-          if (uri) {
-            setAvatarUri(uri);
+  useEffect(() => {
+    let mounted = true;
+
+    const fetchAvatarUri = async () => {
```

Comment:
i learned recently that there's a more optimal way to do this,

useSWR and cache the key to the address avatar, it will reduce the requests on the app considerably, 

Lets do this in a new PR @hernandoagf , once this is merged